lib/memcached.js

```diff
@@ -512,7 +512,12 @@ Client.config = {
 
       switch (flag) {
         case FLAG_JSON:
-          dataSet = JSON.parse(dataSet);
+          try{
+            dataSet = JSON.parse(dataSet);  
+          }catch(e){
+            console.log('Invalid JSON data', e, dataSet);
```

Comment:
Failing silently can cause lots of head-scratching for the users of the library too if something goes wrong...

Maybe passing a flag (like verbose=true or something like that) or a custom logger object (that can have the log level disabled) in the client constructor would be a more appropriate alternative?